File: app-backend/src/main/java/dev/andrei/app_backend/dto/LocationDto.java
package dev.andrei.app_backend.dto;

import java.util.List;
import java.util.UUID;

public record LocationDto(UUID id, String name, String primaryCategory,
                          // UI-friendly category label; null -> client prettifies primaryCategory.
                          String primaryCategoryDisplayName,
                          double longitude, double latitude,boolean hasToilets,
                          boolean hasAccessibilityFeatures, double averageScore, List<AttributeDto> attributes,
                          // Per-user match score (0–5) when ranked by preferences; null otherwise.
                          Double matchScore) {}

File: appfrontend/app/src/main/java/dev/andrei/app_frontend/data/remote/dto/LocationDto.kt
package dev.andrei.app_frontend.data.remote.dto

import dev.andrei.app_frontend.data.local.entity.LocationEntity
import java.util.UUID

data class LocationDto (
    val id: UUID,
    val name: String,
    val primaryCategory: String,
    val primaryCategoryDisplayName: String? = null,
    val longitude: Double,
    val latitude: Double,
    val hasAccessibleFeatures: Boolean,
    val hasToilets: Boolean,
    val averageScore: Double,
    val attributes: List<AttributeDto>,
    val matchScore: Double? = null
){
    fun toEntity() = LocationEntity(id, name, primaryCategory, primaryCategoryDisplayName, longitude, latitude, hasAccessibleFeatures, hasToilets, averageScore, attributes, matchScore)
}